app/src/main/java/com/example/ar_reshare/ToDataURI.java

```diff
@@ -0,0 +1,55 @@
+package com.example.ar_reshare;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.Base64;
+
+public class ToDataURI {
+    public static String TranslateToDataURI(Context context, Uri uri) throws FileNotFoundException {
+
+        InputStream iStream = context.getContentResolver().openInputStream(uri);
+        byte[] inputData = getBytes(iStream);
+        String base64str = Base64.getEncoder().encodeToString(inputData);
+        ContentResolver cR = context.getContentResolver();
+        MimeTypeMap mime = MimeTypeMap.getSingleton();
+        String contentType = mime.getExtensionFromMimeType(cR.getType(uri));
+        // cretate "data URI"
+        StringBuilder sb = new StringBuilder();
+        sb.append("data:");
+        sb.append("image/");
+        sb.append(contentType);
+        sb.append(";base64,");
+        sb.append(base64str);
+        return sb.toString();
+    }
+
+    private static byte[] getBytes(InputStream iStream) {
+        byte[] bytesResult = null;
+        ByteArrayOutputStream byteBuffer = new ByteArrayOutputStream();
+        int bufferSize = 102400000;
```

Comment:
Using a constant here would help!

Reply:
I researched about it and it turns out the buffer size doesn't matter because [ByteArrayOutputStream]will allow for writing to a dynamic byte array. 
So i would change it to 1024